core/plugins/org.polarsys.capella.core.sirius.analysis/src/org/polarsys/capella/core/sirius/analysis/TitleBlockServices.java

```diff
@@ -0,0 +1,515 @@
+          if (feature != null) {
+            EObject objToEvaluate = diagram;
+            List<EObject> modelElements = annotation.getReferences().parallelStream()
+                .filter(x -> !(x instanceof DAnnotation)).collect(Collectors.toList());
+            if (!modelElements.isEmpty()) {
+              objToEvaluate = modelElements.get(0);
+            }
+            Object obj = interpreter.evaluate(objToEvaluate, feature);
+            if (obj != null) {
+              if (obj instanceof EObject) {
+                list.add(obj);
+              } else if (obj instanceof String) {
+                DAnnotation annotationContent = DescriptionFactory.eINSTANCE.createDAnnotation();
+                annotationContent.setSource("abc");
+                annotationContent.getDetails().put(CONTENT, (String) obj);
+                list.add(annotationContent);
+              }
+            } else {
+              DRepresentation representation = (DRepresentation) diagram;
+              list = representation.getOwnedRepresentationElements().stream()
+                  .filter(x -> (x.getTarget() instanceof ClassImpl)).collect(Collectors.toList());
+            }
+          }
+        } catch (EvaluationException e) {
+          e.printStackTrace();
+        }
+      }
+    }
+    return list;
+  }
+
+  public String getTitleBlockCellLabel(EObject diagram, EObject cell) {
+    init();
+    if (cell instanceof DAnnotation) {
+      String name = ((DAnnotation) cell).getDetails().get(NAME);
+      if (name != null)
+        return name;
+    }
+    return "";
+  }
+
+  public String getCellLabel(EObject obj) {
+    if (obj instanceof DAnnotation) {
+      DAnnotation annotation = (DAnnotation) obj;
+      return annotation.getDetails().get(CONTENT);
+    }
+    return "";
+  }
+
+  public boolean isAnnotation(EObject obj) {
+    return obj instanceof DAnnotation;
+  }
+
+  public boolean isTitleBlockAllowed(EObject element, EObject containerView) {
+    if (containerView instanceof DRepresentation) {
+      return true;
+    }
+    return false;
+  }
+
+  public List<DAnnotation> getAvailableTitleBlocksToInsert(final EObject elementView, boolean isDiagramTitleBlock) {
+    List<DAnnotation> result = new ArrayList<>();
+    EList<DAnnotation> eList = ((DDiagram) elementView).getEAnnotations();
+    for (DAnnotation elem : eList) {
+      if (elem.getSource().equals(TITLE_BLOCK)) {
+        result.add(elem);
+      }
+    }
+    List<DAnnotation> finalResult = new ArrayList<>();
+    for (DAnnotation annotation : result) {
+      if (isDiagramTitleBlock) {
+        if (isDiagramTitleBlock(annotation)) {
+          finalResult.add(annotation);
+        }
+      } else {
+        if (!isDiagramTitleBlock(annotation)) {
+          finalResult.add(annotation);
+        }
+      }
+    }
+    return finalResult;
+  }
+
+  public List<DAnnotation> getInitialSelectionOfShowHideTitleBlocks(final EObject elementView,

```

Comment:
same remark as for getAvailableTitleBlocks, create 2 functions one for Diagram TB one for Element TB.

Reply:
Done.